rasa/core/brokers/kafka.py

```diff
@@ -210,15 +216,22 @@ def _publish(self, event: Dict[Text, Any]) -> None:
         )
 
         if self.producer is not None:
-            self.producer.send(
+            self.producer.produce(
```

Comment:
in the confluent examples, they `poll` before producing to fetch callbacks - do we need to do that as well? https://github.com/confluentinc/confluent-kafka-python/blob/master/examples/sasl_producer.py#L86

Reply:
This might actually be a reason why wrapping this in a future and polling on a separate thread could make sense as in this example: https://github.com/confluentinc/confluent-kafka-python/blob/master/examples/asyncio_example.py (as there isn't a natural place for a poll here)

Reply:
Yup will follow the example you shared 👍🏻 to double-check, you're suggesting to follow the implementation of `AIOProducer`?

Reply:
yes I think that would be a good change as it would make sure that any exceptions raised in their callbacks actually end up in the place where we call `produce` and not at a random location in the code where `poll` or `fetch` is called

Reply:
It seems this async implementation leads to the running of integration tests to [hang](https://github.com/RasaHQ/rasa/actions/runs/3534560761/jobs/5931597047). What should I do more in the `finally` stage of the broker test apart from calling `_close()` method? 🤔

Reply:
i managed to fix the issue by calling `producer.list_topics()` as a way to check connection rather than AdminClient, although not sure why this would be a problem. 
Also when I added this [async produce](https://github.com/confluentinc/confluent-kafka-python/blob/e98fa6571c1f3be24b2038bedc06eef922b7f1e2/examples/asyncio_example.py#L53) implementation, integration test was hanging again, so opted to follow the [2nd example](https://github.com/confluentinc/confluent-kafka-python/blob/e98fa6571c1f3be24b2038bedc06eef922b7f1e2/examples/asyncio_example.py#L88): https://github.com/RasaHQ/rasa/pull/11765/commits/48b98b1e8a49e0fa49774546dc8f941037d68e98